core/templates/pages/exploration-editor-page/feedback-tab/services/thread-data-backend-api.service.ts

```diff
@@ -165,45 +175,51 @@ export class ThreadDataBackendApiService {
         let feedbackThreadBackendDicts = threadData.feedback_thread_dicts;
         let suggestionThreadBackendDicts = threadData.suggestion_thread_dicts;
 
-        let suggestionBackendDictsByThreadId = new Map(
-          suggestionBackendDicts.map(dict => [
-            this.suggestionsService.getThreadIdFromSuggestionBackendDict(dict),
-            dict
-          ]));
+        let suggestionBackendDictsByThreadId: SuggestionBackendDictsByThreadId =
+          new Map(
+            suggestionBackendDicts.map(dict => [
+              this.suggestionsService
+                .getThreadIdFromSuggestionBackendDict(dict),
+              dict
+            ])
+          );
 
         return {
           feedbackThreads: feedbackThreadBackendDicts.map(
             dict => this.setFeedbackThreadFromBackendDict(dict)),
           suggestionThreads: suggestionThreadBackendDicts.map(
             dict => this.setSuggestionThreadFromBackendDicts(
-              dict, suggestionBackendDictsByThreadId.get(
+              dict, <SuggestionBackendDict>suggestionBackendDictsByThreadId.get(
                 (dict === null ? null : dict.thread_id))))
         };
       },
       async() => Promise.reject('Error on retrieving feedback threads.'));
   }
 
-  async getMessagesAsync(thread: SuggestionAndFeedbackThread):
+  // A thread will be 'null' if threadId is invalid.
+  async getMessagesAsync(thread: SuggestionAndFeedbackThread | null):
    Promise<ThreadMessage[]> {
     if (!thread) {
       throw new Error('Trying to update a non-existent thread');
     }
     let threadId = thread.threadId;
 
-    return this.http.get(this.getThreadHandlerUrl(threadId)).toPromise()
-      .then((response: ThreadMessages) => {
-        let threadMessageBackendDicts = response.messages;
-        thread.setMessages(threadMessageBackendDicts.map(
-          m => ThreadMessage.createFromBackendDict(m)));
-        return thread.getMessages();
-      });
+    return this.http.get<ThreadMessages>(
+      this.getThreadHandlerUrl(threadId)
+    ).toPromise().then((response: ThreadMessages) => {
+      let threadMessageBackendDicts = response.messages;
+      thread.setMessages(threadMessageBackendDicts.map(
+        m => ThreadMessage.createFromBackendDict(m)));
+      return thread.getMessages();
+    });
   }
 
   async getOpenThreadsCountAsync(): Promise<number> {
-    return this.http.get(this.getFeedbackStatsHandlerUrl()).toPromise()
-      .then((response: NumberOfOpenThreads) => {
-        return this.openThreadsCount = response.num_open_threads;
-      });
+    return this.http.get<NumberOfOpenThreads>(
+      this.getFeedbackStatsHandlerUrl()
+    ).toPromise().then((response: NumberOfOpenThreads) => {
+      return this.openThreadsCount = response.num_open_threads;
```

Comment:
Wait I just saw this code, it is pretty confusing, I know you didn't wrote it, but can you try to fix it?

Reply:
Tried to make it clear, PTAL